# TASK 1 : Netflix Data Clening
## Objective
The objective of this task is to clean the Netflix dataset by removing duplicate records, handling missing values, and standardizing important columns.

# Step 1 : Importing pandas
Import pandas with alias as pd

In [38]:
import pandas as pd

# Step 2 : Loading Dataset
load the dataset and anlyze the first five rows

In [39]:
df = pd.read_csv("Dataset.csv")
df.head()

,show_id,type,title,director,country,date_added,release_year,rating,duration,listed_in
0,s1,Movie,Dick Johnson Is Dead,Kirsten Johnson,United States,9/25/2021,2020,PG-13,90 min,Documentaries
1,s3,TV Show,Ganglands,Julien Leclercq,France,9/24/2021,2021,TV-MA,1 Season,"Crime TV Shows, International TV Shows, TV Act..."
2,s6,TV Show,Midnight Mass,Mike Flanagan,United States,9/24/2021,2021,TV-MA,1 Season,"TV Dramas, TV Horror, TV Mysteries"
3,s14,Movie,Confessions of an Invisible Girl,Bruno Garotti,Brazil,9/22/2021,2021,TV-PG,91 min,"Children & Family Movies, Comedies"
4,s8,Movie,Sankofa,Haile Gerima,United States,9/24/2021,1993,TV-MA,125 min,"Dramas, Independent Movies, International Movies"


# Step 3: Explore the Dataset

Before cleaning the dataset, it is important to understand its structure. We will examine the number of rows and columns, data types, and basic information about the dataset. 

check rows and columns OUTPUT = (rows, columns)

In [40]:
df.shape

(8790, 10)

check column names

In [41]:
df.columns

Index(['show_id', 'type', 'title', 'director', 'country', 'date_added',
       'release_year', 'rating', 'duration', 'listed_in'],
      dtype='str')

check datatypes 

In [42]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 8790 entries, 0 to 8789
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   show_id       8790 non-null   str  
 1   type          8790 non-null   str  
 2   title         8790 non-null   str  
 3   director      8790 non-null   str  
 4   country       8790 non-null   str  
 5   date_added    8790 non-null   str  
 6   release_year  8790 non-null   int64
 7   rating        8790 non-null   str  
 8   duration      8790 non-null   str  
 9   listed_in     8790 non-null   str  
dtypes: int64(1), str(9)
memory usage: 1.6 MB


check null values

In [43]:
df.isnull().sum()

show_id         0
type            0
title           0
director        0
country         0
date_added      0
release_year    0
rating          0
duration        0
listed_in       0
dtype: int64

check duplicates

In [44]:
df.duplicated().sum()

np.int64(0)

# Initial Findings

- Checked the dataset dimensions.
- Examined the column names and data types.
- Checked for missing values.
- Checked for duplicate records.

These observations will guide the data cleaning process.

# Step 4: Initial Dataset Analysis

The dataset was explored to understand its structure before performing any cleaning operations.

### Observations

- Total Records (Rows): **8790**
- Total Features (Columns): **10**
- All columns contain complete data with **no missing values**.
- The dataset contains **no duplicate records**.
- The dataset consists of **9 string columns** and **1 integer column**.

# Step 5: Standardize Text Columns

Text columns may contain unnecessary leading or trailing spaces. These spaces can affect filtering, grouping, and analysis.

The `strip()` function is used to remove extra spaces from the beginning and end of each text value.

In [45]:
df["type"].unique()

<ArrowStringArray>
['Movie', 'TV Show']
Length: 2, dtype: str

In [46]:
df["rating"].unique()

<ArrowStringArray>
[   'PG-13',    'TV-MA',    'TV-PG',    'TV-14',    'TV-Y7',     'TV-Y',
       'PG',     'TV-G',        'R',        'G',    'NC-17',       'NR',
 'TV-Y7-FV',       'UR']
Length: 14, dtype: str

In [47]:
df["country"].head()

0    United States
1           France
2    United States
3           Brazil
4    United States
Name: country, dtype: str

# Step 6: Export the Cleaned Dataset

After completing the data cleaning process, the cleaned dataset is exported as a new CSV file. This ensures that the original dataset remains unchanged while providing a cleaned version for future analysis.

In [50]:
df.to_csv("cleaned_netflix_dataset.csv", index= False)